feat: migrate from Circle CI to GitHub Actions

colincasey opened the pull request:
[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

edmorley commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,19 @@
+name: Test
+on: [push]
```
Same feedback as https://github.com/forcedotcom/sf-fx-runtime-nodejs/pull/436#discussion_r1012065172

```suggestion
on:
  push:
    # Avoid duplicate builds on PRs.
    branches:
      - main
  pull_request:
permissions:
  contents: read
```

joshwlewis replied:
We probably want to follow the pattern we have in our other repos:

```suggestion
on:
  push:
    branches: ["main"]
  pull_request:

permissions:
  contents: read

```

edmorley commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,19 @@
+          cache: 'yarn'
+      - run: yarn install --immutable
+      - run: yarn build
+      - run: yarn lint
```
Missing newline at end of file (hence the icon)

edmorley commented on .circleci/config.yml:
```diff
@@ -1,32 +0,0 @@
-# Javascript Node CircleCI 2.0 configuration file
```
The `.circleci` directory reference in `.npmignore` can now be removed (and replaced by a `.github` entry):
https://github.com/forcedotcom/sf-fx-sdk-nodejs/blob/1d10c6a0cdc3ec76a9e697b1fc6079ee167bbca9/.npmignore#L2

colincasey replied:
i'm going to leave this as is for a couple of reasons:
- `.npmignore` shouldn't even be necessary since the `files` field is used in `package.json` so there should be no worries about publishing the `.github` folder
- we'll no longer be publishing any artifacts from this repository once #146 is merged

edmorley commented on .circleci/config.yml:
```diff
@@ -1,32 +0,0 @@
-          name: Run linter
-          command: npm run lint
-
-      - run:
-          name: Test types are built
-          command: cd fixtures/sample-typescript-project && npm install && npm run build && npm run lint && npm run test
```
This step isn't present in the new GitHub Actions workflow. Is that intentional? I presume not, since it's removal is not mentioned in the PR description.

colincasey replied:
it didn't seem necessary because this repo is going away.  there will be no need to validate these types going forward and the value of this test is questionable as well.

edmorley commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,19 @@
+name: Test
```
It doesn't bother me, so only mentioning in case it does you, but the job name here is `Test`, whereas that for the sibling runtime repo is `Build`, even though the workflows themselves are very similar.

https://github.com/forcedotcom/sf-fx-runtime-nodejs/pull/436/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R1

colincasey replied:
it does bother me but i was using this repository as a light introduction to GitHub Actions so i didn't clue into how this showed up in the UI initially.  i was going to leave it though since this repo will no longer require building after #146 is merged.

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)
